File: src/test/java/ch/grassl/day/day08/GridTest.java
package ch.grassl.day.day08;

import ch.grassl.day.day08.impl.Grid;
import ch.grassl.day.day08.impl.GridImporter;
import ch.grassl.day.day08.impl.Tree;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class GridTest {

    private final GridImporter importer = new GridImporter();

    @Test
    void testGrid_TreeNeighbours_1() {
        Grid grid = importer.importData(GridImporterTest.RESOURCE);
        Tree tree = grid.getTrees()[0][3];
        assertEquals(1, tree.getSouth().getSize());
        assertEquals(3, tree.getEast().getSize());
        assertEquals(3, tree.getWest().getSize());
        assertTrue(tree.isVisible());
    }

    @Test
    void testGrid_TreeNeighbours_2() {
        Grid grid = importer.importData(GridImporterTest.RESOURCE);
        Tree tree = grid.getTrees()[2][2];
        assertEquals(5, tree.getSouth().getSize());
        assertEquals(3, tree.getEast().getSize());
        assertEquals(5, tree.getWest().getSize());
        assertEquals(5, tree.getNorth().getSize());
        assertFalse(tree.isVisible());
    }

    @Test
    void testGrid_TreeNeighbours_3() {
        Grid grid = importer.importData(GridImporterTest.RESOURCE);
        Tree tree = grid.getTrees()[4][4];
        assertEquals(9, tree.getNorth().getSize());
        assertEquals(9, tree.getWest().getSize());
        assertTrue(tree.isVisible());
    }

    @Test
    void testGrid_NumberOfVisibleTrees() {
        Grid grid = importer.importData(GridImporterTest.RESOURCE);
        assertEquals(21, grid.numberOfVisibleTrees());
    }

    @Test
    void testGrid_ScenicScore() {
        Grid grid = importer.importData(GridImporterTest.RESOURCE);
        assertEquals(8, grid.findHighestScenicScore());
    }
}

File: src/test/java/ch/grassl/day/day09/impl/GridTest.kt
package ch.grassl.day.day09.impl

import ch.grassl.day.day09.impl.importer.MoveImporter
import ch.grassl.day.day09.impl.model.State
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

internal class GridTest {

    private val moves = MoveImporter().importData(MoveImporterTest.RESOURCE)
    private val gridSize = 20

    @Test
    fun testInit() {
        val grid = Grid(moves, gridSize)
        assertEquals(gridSize, grid.getFields().size)
        assertEquals(gridSize, grid.getFields()[0].size)
    }

    @Test
    fun testPrint() {
        val grid = Grid(moves, gridSize)
        print(grid)
        assertEquals(State.START, grid.getFields()[grid.getFields().size / 2][grid.getFields().size / 2].getState())
    }


    @Test
    fun testApplyMoves() {
        val grid = Grid(moves, gridSize)
        grid.applyMoves(true)
    }

    @Test
    fun testCountVisited() {
        val grid = Grid(moves, gridSize)
        grid.applyMoves()
        assertEquals(14, grid.countVisited())
    }
}